corehq/apps/users/models.py

```diff
@@ -1216,6 +1214,7 @@ def clear_quickcache_for_user(self):
 
         self.get_by_username.clear(self.__class__, self.username)
         self.get_by_user_id.clear(self.__class__, self.user_id)
+        username_to_user_id.clear(self.username)
```

Comment:
There's also no reason we couldn't call `user.clear_quickcache_for_user()` for each user during bulk upload if we wanted to make sure that bulk uploads don't create an inconsistent cache situation.